src/Coalesce.Services/Search/service-data/src/main/java/com/incadencecorp/coalesce/services/search/service/data/controllers/TemplateDataController.java

```diff
@@ -321,12 +321,23 @@ public String setTemplate(String key, CoalesceEntity entity) throws RemoteExcept
                                     e.getMessage()), e);
             }
 
+            if (!key.equals("new"));
+            {
+                // We want to make sure the template key is hashed from name/source/version, not arbitrary.
+                template.setKey(null);
+                generatedKey = template.getKey();
```

Comment:
This is the only place generatedKey is being set so if the key == null then this method will return null.